Work in progress on Entities...

aespasa opened the pull request:
Create Entities and implements validator on them.

Create database structure SQL script.

Refactor code style and docs.

ObjectHydrator now support Column's Date Type.

aespasa committed:
Create Entities and implements validator on them.

Create database structure SQL script.

Refactor code style and docs.

ObjectHydrator now support Column's Date Type.

TBoileau commented on src/Entity/Post.php:
```diff
@@ -10,36 +10,53 @@
-    #[Column(name: 'id', unique: true)]
+    #[Assert\Uuid()]
     protected string $id;
 
```
Il manque l'image

aespasa committed:
Refactor Constraints and Constraints validation

aespasa committed:
Repair registerExisting container's method and add Unique constraint with its validator.

Also repair some bugs

TBoileau commented on src/DependencyInjection/Container.php:
```diff
@@ -83,6 +83,7 @@ function (ReflectionParameter $parameter) {
     public function getDefinition(string $id): Definition
     {
         if (!isset($this->definitions[$id])) {
+            dump($id);
```
A supprimer

TBoileau commented on src/Entity/User.php:
```diff
@@ -21,26 +21,26 @@ class User
     #[Assert\MaxLength(
         message: "Le nom d'utilisateur '%s' ne peut excéder %d caractères. Il en possède actuellement %d",
-        maxLen: 20)
+        maxLength: 20)
```
ça fait un peu redondant, renommer "max"

TBoileau commented on src/Entity/User.php:
```diff
@@ -21,26 +21,26 @@ class User
     ]
     #[Assert\MinLength(
         message: "Le nom d'utilisateur '%s' doit faire au moins %d caractères. Il en possède actuellement %d",
         minLength: 3)
```
=> "min"

TBoileau commented on src/Validator/Constraint/Constraint.php:
```diff
@@ -4,14 +4,13 @@
 
-    public function getMessage(): string
+    public static function getInstance(): ConstraintInterface
     {
-        return $this->message;
+        $constraint = static::class;
+        return new $constraint();
     }
```
j'ai du mal à comprendre l'intérêt d'avoir une factory method

aespasa committed:
Refactor code

aespasa committed:
Refactor constraints, validators and Request handling. Create a FileUploader service. Refactor object handling from the hydrator.

A constraint attribute now get its own constraint validator which is called automatically by the Validator. It's useful to initialize some required objects from the constructor (like the MySQLAdapter in the Unique constraint validation).

A FileUploader is now used to upload or replace a file.

The main Request is now initialized when configuring the DIC: In this way, we can retrieve the main request everywhere we need it (like when generating the csrf_token from the Csrf Twig Extension).

Add many tests.

aespasa committed:
Refactor some code and work on the FrontController (Authentication)

aespasa committed:
Work on login feature

aespasa committed:
Add PSR-14 Event Dispatcher

aespasa committed:
Add flash messages display and split Security features from the FrontController to a SecurityController

aespasa committed:
Refactor the ObjectHydrator to read columns name when getting a resource from the database. Implement login system with an Authenticator. Create a PSR-14 Event Dispatcher.

aespasa committed:
Add authenticator, login system with remember me cookie, and add fixtures

aespasa committed:
Wip

aespasa committed:
Refactor Route to remove ?queries. Now load Event Listeners through the DIC. Refactor some code.

aespasa committed:
Implement pagination and pagination range
